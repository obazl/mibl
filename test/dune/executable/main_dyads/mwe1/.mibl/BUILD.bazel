load("@bazel_skylib//rules:diff_test.bzl", "diff_test")

diff_test(
    name  = "obazl_test",
    file1 = "BUILD.bazel.expected",
    file2 = "//test/dune/executable/main_dyads/mwe1:BUILD.bazel",
    timeout = "short"
)

sh_test(
    name = "mibl_test",
    srcs = ["//test:Mibl_test.sh"],
    deps = ["@bazel_tools//tools/bash/runfiles"],
    data = ["//test:mibl_test", "PARSETREE.s7", "PROJECT.s7"],
    timeout = "short"
)

