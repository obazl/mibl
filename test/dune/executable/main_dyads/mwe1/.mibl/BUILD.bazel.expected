## GENERATED FILE ##

exports_files(["BUILD.bazel"]) # for testing

load("@rules_ocaml//build:rules.bzl",
     "ocaml_binary",
     "ocaml_exec_module",
     "ocaml_library",
     "ocaml_module",
     "ocaml_ns_library",
     "ocaml_ns_resolver",
     "ocaml_signature")

##############################
####  Executable Targets  ####
ocaml_binary(
    name     = "main1.exe",
    main     = ":Main1",
    visibility = ["//visibility:public"],
)

ocaml_exec_module(
    name          = "Main1",
    struct        = "main1.ml",
    sig           = ":Main1_cmi",
)

ocaml_signature(
    name          = "Main1_cmi",
    src           = "main1.mli",
)


ocaml_binary(
    name     = "main2.exe",
    main     = ":Main2",
    visibility = ["//visibility:public"],
)

ocaml_exec_module(
    name          = "Main2",
    struct        = "main2.ml",
    sig           = ":Main2_cmi",
)

ocaml_signature(
    name          = "Main2_cmi",
    src           = "main2.mli",
)


######################## Modules & Signatures ########################
